masonry/src/event.rs

```diff
@@ -50,6 +50,84 @@ pub enum PointerButton {
     Other,
 }
 
+/// A set of [`PointerButton`]s.
+#[derive(PartialEq, Eq, Clone, Copy, Default)]
+pub struct PointerButtons(u8);
+
+fn button_bit(button: PointerButton) -> u8 {
+    match button {
+        PointerButton::None => 0,
+        PointerButton::Primary => 0b1,
+        PointerButton::Secondary => 0b10,
+        PointerButton::Auxiliary => 0b100,
+        PointerButton::X1 => 0b1000,
+        PointerButton::X2 => 0b10000,
+        // TODO: When we properly do `Other`, this changes
+        PointerButton::Other => 0b100000,
+    }
+}
+
+impl PointerButtons {
+    /// Create a new empty set.
+    #[inline]
+    pub fn new() -> PointerButtons {
+        PointerButtons(0)
+    }
+
+    /// Add the `button` to the set.
+    #[inline]
+    pub fn insert(&mut self, button: PointerButton) {
+        self.0 |= button_bit(button);
+    }
+
+    /// Remove the `button` from the set.
+    #[inline]
+    pub fn remove(&mut self, button: PointerButton) {
+        self.0 &= !button_bit(button);
+    }
+
+    /// Returns `true` if the `button` is in the set.
+    #[inline]
+    pub fn contains(self, button: PointerButton) -> bool {
+        (self.0 & button_bit(button)) != 0
+    }
+
+    /// Returns `true` if the set is empty.
+    #[inline]
+    pub fn is_empty(self) -> bool {
+        self.0 == 0
+    }
+
+    /// Returns `true` if all the `buttons` are in the set.
+    #[inline]
+    pub fn is_superset(self, buttons: PointerButtons) -> bool {
+        self.0 & buttons.0 == buttons.0
+    }
+
+    /// Adds all the `buttons` to the set.
+    pub fn extend(&mut self, buttons: PointerButtons) {
+        self.0 |= buttons.0;
+    }
+
+    /// Clear the set.
+    #[inline]
+    pub fn clear(&mut self) {
+        self.0 = 0;
+    }
+
+    /// Count the number of pressed buttons in the set.
+    #[inline]
+    pub fn count(self) -> u32 {
+        self.0.count_ones()
+    }
+}
+
+impl std::fmt::Debug for PointerButtons {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "PointerButtons({:05b})", self.0)
```

Comment:
Can we display this in full? I.e. `Primary | Secondary | ...`

We can maybe implement [`std::fmt::Binary`](https://doc.rust-lang.org/std/fmt/trait.Binary.html) to use this display?

Reply:
This is done now as well.